src/syngen/ml/vae/models/dataset.py

```diff
@@ -407,11 +429,29 @@ def _set_date_columns(self, df: pd.DataFrame):
             get_date_columns(df, list(self.str_columns)) - self.categ_columns - \
             self.binary_columns - self.long_text_columns
 
-    def _set_empty_columns(self, df: pd.DataFrame, schema: Dict):
+    def _set_empty_columns(self):
         """
         Set up the list of empty columns which have been dropped from the table
         """
-        self.empty_columns = [column for column in schema["fields"] if schema["fields"][column] == "removed"]
+        self.empty_columns = {
+            column for column in self.schema["fields"]
+            if self.schema["fields"][column] == "removed"
+        }
+
+    def _set_non_existent_columns(self):
```

Comment:
Why do we need to store these columns? Can we just drop them from metadata and proceed (after a warning)?